# DH 140 Assignment 2

Spurthi Rallapalli, DH 140 W22

## **Description**: In this assignment, I will be looking at and analyzing some (Python) earthquake data from the CORGIS datasets, found at [this link](https://corgis-edu.github.io/corgis/python/earthquakes/).

## Reading the Datafile

First, I will be **reading the datafile**. Looking at the earthquakes.py file, I found that I could use `get_earthquake()` to read the data, similar to what we did in class (Week 2).

In [1]:
# From lecture, I saw that we need to import earthquakes first
import earthquakes

In [2]:
# Then, use the get_earthquake() function to get the data
earthquake_data = earthquakes.get_earthquake()

Now, let's look at the type and length of this data.

In [3]:
type(earthquake_data)

list

In [4]:
len(earthquake_data)

8394

*So, the data provided is of type list, and there is information about 8,394 earthquakes provided.*

Now, I want more information about the information that this dataset provides, so I'm going to print out the very first earthquake data.

In [5]:
earthquake_data[0]

{'id': 'nc72666881',
 'impact': {'gap': 122.0, 'magnitude': 1.43, 'significance': 31},
 'location': {'depth': 15.12,
  'distance': 0.1034,
  'full': '13km E of Livermore, California',
  'latitude': 37.6723333,
  'longitude': -121.619,
  'name': 'California'},
 'time': {'day': 27,
  'epoch': 1469593183550,
  'full': '2016-07-27 00:19:43',
  'hour': 0,
  'minute': 19,
  'month': 7,
  'second': 43,
  'year': 2016}}

*From above, we can see that the datafile has id, impact, location, and time fields, each with subfields.*

## Mathematical Operation

Now, I want to try implementing a **mathematical operation**. For this very first data value (shown above), I will multiply the depth value by 1000 to change the units from km to m.

Note: According to [this webpage](https://www.usgs.gov/programs/earthquake-hazards/determining-depth-earthquake), the depth of an earthquake is measured in kilometers. 

In [7]:
depth_meters = earthquake_data[0]['location']['depth'] * 1000
print("Depth in meters is: " + str(depth_meters) + " meters.")

Depth in meters is: 15120.0 meters.


*So we see using a mathematic operation that the depth of this specific earthquake is 15120 meters, or 15.12 km.*

Now, I'm going to experiment with **conditional expressions**. According to [this website](https://www.usgs.gov/programs/earthquake-hazards/determining-depth-earthquake), also listed in the mathematical operations section, mentions that "Shallow earthquakes are between 0 and 70 km deep; intermediate earthquakes, 70 - 300 km deep; and deep earthquakes, 300 - 700 km deep", which I am going to apply to the first data value.

In [8]:
#use if statement and conditionals, using the following information
# 0->70 km = Shallow
# 70->300 km = Intermediate
# 300->700 k = Deep

if (earthquake_data[0]['location']['depth'] < 70):
    print("This is a shallow earthquake.")
elif (earthquake_data[0]['location']['depth'] < 300):
    print("This is an intermediate earthquake")
else:
    print("This is an deep earthquake")

This is a shallow earthquake.


*As predicted, the above printed that the earthquake we were looking at was a shallow earthquake, because it is only 15.12 km deep.*

## For Loop

In the following section, I'm going to use a **for loop** to state the type of every earthquake in the dataset.

In [9]:
#look through every piece of earthquake data... should produce 8394 lines of classifications.

for i in earthquake_data:
    if (i['location']['depth'] < 70):
        print("This is a shallow earthquake.")
    elif (i['location']['depth'] < 300):
        print("This is an intermediate earthquake")
    else:
        print("This is an deep earthquake")

This is a shallow earthquake.
This is an intermediate earthquake
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is an intermediate earthquake
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is a shallow earthquake.
This is an intermediate earthquake
This is an intermediate earthquake
This is a shallow ea

*As we can see printed above, there are a lot of shallow and some intermediate earthquakes in this dataset*

## Functions

I'm going to implement something similar to the for loop functionality above, but using a **function**. In this function, I will only print if an earthquake is an intermediate earthquake, along with it's location (full), and its depth. 

In [13]:
def print_intermediate():
    for i in earthquake_data:
        if (i['location']['depth'] < 300 and i['location']['depth'] >= 70):
            print("This is an intermediate earthquake with depth " + str(i['location']['depth']) + " located at " + i['location']['full'])

In [14]:
print_intermediate()

This is an intermediate earthquake with depth 97.07 located at 58km WNW of Pakokku, Burma
This is an intermediate earthquake with depth 70.5 located at 78km SW of Homer, Alaska
This is an intermediate earthquake with depth 98.84 located at 155km SSE of San Pedro de Atacama, Chile
This is an intermediate earthquake with depth 163.0 located at 4km W of Salvaleon de Higuey, Dominican Republic
This is an intermediate earthquake with depth 72.2 located at 32km W of Kalifornsky, Alaska
This is an intermediate earthquake with depth 75.0 located at 114km NNE of Road Town, British Virgin Islands
This is an intermediate earthquake with depth 234.15 located at 42km NNE of Poigar, Indonesia
This is an intermediate earthquake with depth 95.9 located at 96km W of Willow, Alaska
This is an intermediate earthquake with depth 84.4 located at 57km SW of Anchor Point, Alaska
This is an intermediate earthquake with depth 71.0 located at 95km NW of Kodiak Station, Alaska
This is an intermediate earthquake 

## Module Imports

Finally, I will make use of module imports, more specifically, the math module from lecture. I found this function called `math.ceil(x)` that I'll use to take the depth of the first data value and round up.

In [16]:
import math

In [17]:
math.ceil(earthquake_data[0]['location']['depth'])

16

*This worked as expected, 15.12 km was rounded up to 16 km*

## That's it for my analysis of this dataset for Assignment 2! Thank you!